lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -163,12 +176,91 @@ export class AuthorizationCodeModule extends AuthModule {
         // Create response object
         const response: CodeResponse = {
             code: hashParams.code,
-            userRequestState: ProtocolUtils.getUserRequestState(hashParams.state)
+            userRequestState: hashParams.state
         };
 
         return response;
     }
 
+    private createTokenResponse(serverTokenResponse: ServerAuthorizationTokenResponse, state: string): TokenResponse {
```

Comment:
Should Token response just be a class that takes ServerAuthorizationTokenResponse as a constructor ?

Reply:
We return the `TokenResponse` type to the user (i.e. what we return today as the `AuthResponse` type), so I can make a separate class that does what you describe, just was a little lost around naming.

Reply:
done, created a response handler class